### PR TITLE
feat(parquet):support RLE encoding

### PR DESCRIPTION
**Pull request by jun0315**
[Parquet2](https://github.com/jorgecarleitao/parquet2#encoding) already supports RLE encoding, we need to add it to the `can_encode` Support for adding RLE encoding to the encode function

### CODE REVIEW

**Review by sundy-li — APPROVED** [review]

**Comment by ritchie46** [comment]
Thanks.. does this work as is? Could we maybe add a test for this.

**Comment by jun0315** [comment]
Now arrow2 isn't support rle encoding
https://github.com/jorgecarleitao/arrow2/blob/b09e580f075293e9af2879dcc7f6b2d5c8fe520e/src/io/parquet/write/primitive/basic.rs#L94-L111

File: tests/it/io/parquet/write.rs [hunk 1]
```diff
@@ -70,12 +71,13 @@ fn round_trip_opt_stats(
...
     let data = writer.into_inner().into_inner();
 
-    let (result, stats) = read_column(&mut Cursor::new(data), "a1")?;
-
-    assert_eq!(array.as_ref(), result.as_ref());
-    if check_stats {
-        assert_eq!(statistics, stats);
-    }
+    fs::write("test.parquet", data).unwrap();
+    // let (result, stats) = read_column(&mut Cursor::new(data), "a1")?;
```
Comment (ritchie46) [inline tests/it/io/parquet/write.rs:75]: Can you uncomment this test?

**Review by ritchie46 — COMMENTED** [review]
Thanks, Almost there. There was some code commented.

**Pull request closed without merging**